src/anomalib/utils/attrs.py

```diff
@@ -0,0 +1,24 @@
+"""Utility functions for working with attributes."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any
+
+
+def get_nested_attr(obj: Any, attr_path: str, default: Any | None = None) -> Any:  # noqa: ANN401
```

Comment:
Can you elaborate why this is needed? Maybe add more description to the docstring?